server/src/main/kotlin/org/javacs/kt/index/SymbolIndex.kt

```diff
@@ -1,45 +1,79 @@
 package org.javacs.kt.index
 
-import org.jetbrains.exposed.sql.and
-import org.jetbrains.exposed.sql.count
-import org.jetbrains.exposed.sql.deleteAll
-import org.jetbrains.exposed.sql.innerJoin
-import org.jetbrains.exposed.sql.replace
-import org.jetbrains.exposed.sql.select
-import org.jetbrains.exposed.sql.selectAll
-import org.jetbrains.exposed.sql.Table
 import org.jetbrains.exposed.sql.transactions.transaction
-import org.jetbrains.exposed.sql.SchemaUtils
 import org.jetbrains.kotlin.descriptors.ModuleDescriptor
 import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
 import org.jetbrains.kotlin.resolve.scopes.DescriptorKindFilter
-import org.jetbrains.kotlin.resolve.scopes.MemberScope
 import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
 import org.jetbrains.kotlin.name.FqName
-import org.jetbrains.kotlin.psi2ir.intermediate.extensionReceiverType
 import org.javacs.kt.LOG
 import org.javacs.kt.progress.Progress
-import org.jetbrains.exposed.sql.Database
-import org.jetbrains.exposed.sql.SqlExpressionBuilder.like
-import org.jetbrains.exposed.sql.insert
-
-private val MAX_FQNAME_LENGTH = 255
-private val MAX_SHORT_NAME_LENGTH = 80
-
-private object Symbols : Table() {
-    val fqName = varchar("fqname", length = MAX_FQNAME_LENGTH) references FqNames.fqName
+import org.jetbrains.exposed.dao.IntEntity
+import org.jetbrains.exposed.dao.IntEntityClass
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.*
+import java.util.concurrent.CompletableFuture
+import kotlin.sequences.Sequence
+
+private const val MAX_FQNAME_LENGTH = 255
+private const val MAX_SHORT_NAME_LENGTH = 80
+private const val MAX_URI_LENGTH = 511
+
+private object Symbols : IntIdTable() {
```

Comment:
Just to understand, we are now using integers (instead of fully-qualified names) to identify symbols. Can we still be sure that uniqueness is maintained (perhaps by using `.uniqueIndex()` instead of `.index()`)?

Reply:
Right, I didn't use a unique index because I think it would complicate things when a user creates multiple classes with the same fully qualified name. Even though it's illegal (and will lead to compilation errors), it's still technically possible for the user to do this. Therefore, I chose to include the duplicate entries in the index.

We could certainly make it unique, but I think we would need to add some more logic to prevent duplicates. And if we ever start using the index for other things (like definitions, for example) duplicated symbols would be ignored from that.